yt_dlp/extractor/niconico.py

```diff
@@ -1005,6 +1007,7 @@ def _real_extract(self, url):
             if data.get('type') == 'stream':
                 m3u8_url = data['data']['uri']
                 qualities = data['data']['availableQualities']
+                cookies = data['data'].get('cookies', [])
```

Comment:
There won't always be cookies? Then I would do this:

```suggestion
                cookies = traverse_obj(data, ('data', 'cookies', ..., {dict}))
```

** **

Or, if we always *need* cookies, then let it be completely fatal here:
```suggestion
                cookies = data['data']['cookies']
```